File: frontend/src/app/interfaces/filters/AgencyPackageFilter.ts
export interface AgencyPackageFilter {
  name?: string;
  tags?: string[];
  status?: string;
  minPrice?: number;
  maxPrice?: number;
  authorId?: string;
}

export interface FullAgencyPackageFilter extends AgencyPackageFilter {
  to?: string;
  toType?: string;
  startDate?: string;
  endDate?: string;
}

File: bff-api/src/main/java/com/certimetergroup/smart_travel/bff_api/filter/AgencyPackageFilter.java
package com.certimetergroup.smart_travel.bff_api.filter;

import java.time.Instant;
import java.util.Set;
import lombok.Data;
import lombok.NoArgsConstructor;
import shared.PackageStatus;

@Data
@NoArgsConstructor
public class AgencyPackageFilter {

  private String name;
  private String city;
  private String region;
  private String country;
  private Set<String> tags;
  private PackageStatus status;
  private Double minPrice;
  private Double maxPrice;
  private Instant startDate;
  private Instant endDate;
  private String authorId;
}

